### PR TITLE
add in extended documentation for return values Fixes #79

### LINKED ISSUE #79: Expand  documentation on return values
@rawls238  made some improvements to the logging documentation, but there should probably be an entire subsection on return values, noting that (1) it is common for experiments to include some kind of eligibility check, and that you should always return false if users fail the eligibility check, so that they don't get logged (2) returning false is also useful when you are dropping experimental conditions

### External service example
This might include some code example whose `assign()` method includes some kind of check to an external service, e.g.,

```python
country = get_user_country(userid)
if country != '':
  params.feature_enabled = false
  return false
params.feature_enabled = ...
```

### Dropping arm example
Suppose you are running an experiment that tests 4 difference prices for a trade in a prediction market:

```
params.trade_price = uniformChoice(choices=[0.10, 0.25, 0.50, 0.99], unit=userid)
```

and it turns out that 0.99 performs very poorly,  If we were to simply change the experiment to say,

```
params.trade_price = uniformChoice(choices=[0.10, 0.25, 0.50], unit=userid)
```

Doing this can have a number of negative effects: (1) it will reshuffle all users (2) in some cases (particularly with weightedChoice), doing this can create carryover effects, where users get re-shuffled in non-random ways.

One solution would be to simply replace 0.99 with your best guess of what works best (say it's 0.5):

```
params.trade_price = uniformChoice(choices=[0.10, 0.25, 0.50, 0.50], unit=userid)
```

This would break randomization in your experiment because potential carryover effects from those previously in the 0.99 condition could bias your estimates from the 0.50 condition.

To prevent this from happening, you can move all users who were previously in the 0.99 condition over to what you presently believe works best, and then tell PlanOut not to log the outcomes from those users:

```
params.trade_price = uniformChoice(choices=[0.10, 0.25, 0.50, 0.99], unit=userid)
if params.trade_price == 0.99:
  params.trade_price = 0.5
  return false
```

### PR DESCRIPTION
**Pull request by rawls238**
@eytan

This is mostly taken verbatim from your suggested additions to the documentation from #79, but I added a bit around it. Let me know what you think we could add/remove/reword

### CODE REVIEW

File: docs/04-logging.md [hunk 1]
```diff
@@ -59,6 +58,58 @@ checksum can only be computed when Python is running in non-interactive mode,
...
+### Example use when initially creating an experiment
+
+For instance, suppose you are running an experiment on a new feature and want to only target users in Germany for your experiment. By returning `False` from your assignment function for users that are not from Germany, PlanOut ensures that you will only have exposure logs for users that are from Germany and can still fix parameters for the users that are not from Germany. In this case your `assign` function may look something like this if it pulls the current user's country from an external service:
+
+```python
+class Exp1(SimpleExperiment):
+  def assign(self, params, userid):
+    country = get_user_country(userid)
+    if country != 'Germany':
+      params.feature_enabled = false
```
Comment (eytan) [inline docs/04-logging.md]: This is a nice example, but FWIW, it's better style to define all the defaults for your params at the top of your script.

File: docs/04-logging.md [hunk 2]
```diff
@@ -59,6 +58,58 @@ checksum can only be computed when Python is running in non-interactive mode,
...
+
+### Example use when iterating on an experiment
+
+Suppose you are running an experiment that tests 4 difference prices for a trade in a prediction market:
+
+```
+params.trade_price = uniformChoice(choices=[0.10, 0.25, 0.50, 0.99], unit=userid)
+```
+
+and it turns out that 0.99 performs very poorly, If we were to simply change the experiment to say,
```
Comment (eytan) [inline docs/04-logging.md]: s/If/if

**Comment by eytan** [comment]
Could you change those two nits? Otherwise looks good, thanks for contributing this @rawls238 .

**Comment by rawls238** [comment]
@eytan will do that tonight. I have a branch locally that has a few other fixups for the docs that I'll push tonight as well

**Comment by rawls238** [comment]
@eytan should be good to go

**Comment by eytan** [comment]
Somehow I missed this update --- merging now...

**Pull request merged**